scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -160,41 +188,112 @@
     (log/info "In compute cluster" name ", processing pod" pod-name ";"
               "expected:" (prepare-expected-state-dict-for-logging expected-state-dict) ","
               "existing:" (prepare-existing-state-dict-for-logging existing-state-dict))
-    ; TODO: We added an :expected/starting state to the machine, to represent when a pod is starting. We map instance.status/unknown to that state
-    ; The todo is to add in cases for [:expected/starting *] for those other states.
+    ; We should have the cross product of
+    ;      :expected/starting :expected/running :expected/completed :expected/killed :missing
+    ; and
+    ;      :pod/waiting :pod/running :pod/succeeded :pod/failed :pod/unknown :missing
+    ;
+    ; for a total of 30 states.
+    ;
+    ; The only terminal expected state is :expected/completed and :missing
+    ; The only terminal pod state is :pod/succeeded :pod/failed and :missing. We also treat :pod/unknown as a terminal state.
+
```

Comment:
Which combination of states are terminal? is it the cross product of the individual terminal states?

Reply:
Ok.